src/main/java/com/atlassian/db/replica/api/AuroraPostgresLsnReplicaConsistency.java

```diff
@@ -33,16 +38,28 @@ public Builder cacheLastWrite(Cache<Long> lastWrite) {
             return this;
         }
 
+        /**
+         * @param cache
+         */
+        public Builder replicaLsnCache(SuppliedCache<Long> cache) {
```

Comment:
It's a way of injecting a new feature via new API, without affecting old API. And the default is equivalent to old behavior. Very well done.